pipelines/utils/tests/test_upload_to_gcs_safety.py

```
@@ -0,0 +1,117 @@
+        dataset_id="us_sec_edgar",
+        table_id="dicionario",
+    )
+
+    tb.client["bigquery_staging"].update_table.assert_not_called()
+
+
+@patch("pipelines.utils.tasks.dump_header")
+@patch("pipelines.utils.tasks.bd")
+def test_overwrite_never_deletes_prod(bd_mod, dump_header_mock):
+    """O ponto central: `overwrite` só pode apagar staging."""
+    dump_header_mock.return_value = "/tmp/header.parquet"
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/header.parquet"

(S108)